app/src/main/java/com/monstarlab/features/login/LoginFragment.kt

```diff
@@ -36,15 +38,21 @@ class LoginFragment : Fragment(R.layout.fragment_login) {
     }
 
     private fun collectFlows() {
-        collectFlow(viewModel.loginResultFlow) {
-            findNavController().navigate(R.id.resourceFragment)
-        }
-
-        collectFlow(viewModel.loadingFlow) { loading ->
-            TransitionManager.beginDelayedTransition(binding.root)
-            binding.loginEmailEditText.isEnabled = !loading
-            binding.loginPasswordEditText.isEnabled = !loading
-            binding.loginButton.isVisible = !loading
+        repeatWithViewLifecycle {
+            launch {
+                viewModel.loginResultFlow.collect {
+                    findNavController().navigate(R.id.resourceFragment)
+                }
+            }
+
+            launch {
+                viewModel.loadingFlow.collect { loading ->
+                    TransitionManager.beginDelayedTransition(binding.root)
+                    binding.loginEmailEditText.isEnabled = !loading
+                    binding.loginPasswordEditText.isEnabled = !loading
+                    binding.loginButton.isVisible = !loading
+                }
+            }
```

Comment:
Naming aside, I dont see how this is nicer in any way?

Reply:
yeah, this is probably not nicer. but based on the documentation of `flowWithLifecycle`
https://androidx.tech/artifacts/lifecycle/lifecycle-runtime-ktx/2.4.0-alpha01-source/androidx/lifecycle/FlowExt.kt.html
it's better this way
`Tip: If multiple flows need to be collected using flowWithLifecycle, consider using the Lifecycle.repeatOnLifecycle API to collect from all of them using a different launch per flow instead. That's more efficient and consumes less resources as no hot flows are created.`